internal/providers/github/github_app_installation.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v56/github"
+
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// AppInstallationDelegate is the struct that contains the GitHub App specific operations
+type AppInstallationDelegate struct {
```

Comment:
I'm not a big fan of this name; what about GitHubAppProvider?

Reply:
`GitHubAppDelegate` would work. I'm avoiding putting `Provider` in the name because it doesn't implement the GitHub provider, only a subset of the functions on it.